relayer/app/cursor/cursor.go

```diff
@@ -0,0 +1,47 @@
+package cursor
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/xchain"
+
+	ormv1alpha1 "cosmossdk.io/api/cosmos/orm/v1alpha1"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/orm/model/ormdb"
+	db "github.com/cosmos/cosmos-db"
+)
+
+func (c *Cursor) StreamID(shardID xchain.ShardID) xchain.StreamID {
```

Comment:
this isn't great. Providing a shard, and then returning the stream feels like incorrect abstractions. Someone can provide any shardID, not matching the actual conf level. I don't think should be a method on the cursor.